solcore/data_analysis_tools/ellipsometry_analysis.py

```diff
@@ -129,14 +130,14 @@ def plot(self, log=True):
 class EllipsometryFitter:
     """ Class that contains all the tools to fit the properties of a OptiStack model to ellipsometry data."""
 
-    def __init__(self, data, stack=None, vars=None, wavelength_range=None):
+    def __init__(self, data, stack=None, vars=None, wavelength_range=None) -> None:
 
-        self.data = data
-        self.stack = stack
-        self.vars = vars
-        self.range = wavelength_range
+        self.data: EllipsometryData = data
+        self.stack: OptiStack = stack
+        self.vars: Collection = vars
```

Comment:
Given that `stack`, `vars` and  `wavelength_range`, can all be initialised with `None`, the correct type annotation are `Optional[OptiStack]`, `Optional[Collection]` and `Optional[np.ndarray]`, respectively.

Besides, `np.array` is not a type. For NumPy arrays, the type is `np.ndarray`.

Reply:
Actually i noticed that i made a mistake typing in the body of the function instead of the signature of the function, now should be correct.